ste/xfer-setProperties.go

```diff
@@ -0,0 +1,224 @@
+					jptm.Log(pipeline.LogError, errMsg)
+					common.GetLifecycleMgr().Error(errMsg)
+				}
+			}
+
+			// in all other cases, make the transfer as failed
+			transferDone(common.ETransferStatus.Failed(), err)
+		} else {
+			transferDone(common.ETransferStatus.Success(), nil)
+		}
+	}
+
+	if SetPropertiesAPIOption.TransferMetaData() {
+		panic("Not Supported to be setting metadata yet")
+	}
+}
+
+func setPropertiesBlobFS(jptm IJobPartTransferMgr, p pipeline.Pipeline) {
+	info := jptm.Info()
+	// Get the source blob url of blob to delete
+	u, _ := url.Parse(info.Source)
+	srcBlobURL := azbfs.NewFileURL(*u, p)
+
+	// Internal function which checks the transfer status and logs the msg respectively.
+	// Sets the transfer status and Report Transfer as Done.
+	// Internal function is created to avoid redundancy of the above steps from several places in the api.
+	transferDone := func(status common.TransferStatus, err error) {
+		if status == common.ETransferStatus.Failed() {
+			jptm.LogError(info.Source, "SET-PROPERTIES ERROR ", err)
+		} else {
+			jptm.Log(pipeline.LogInfo, fmt.Sprintf("SET-PROPERTIES SUCCESSFUL: %s", strings.Split(info.Destination, "?")[0]))
+		}
+
+		jptm.SetStatus(status)
+		jptm.ReportTransferDone()
+	}
+
+	_, err := srcBlobURL.SetAccessControl(jptm.Context(), azbfs.BlobFSAccessControl{})
+	if err != nil {
+		if strErr, ok := err.(azblob.StorageError); ok {
+			// TODO: Do we need to add more conditions?
+
+			// If the status code was 403, it means there was an authentication error, and we exit.
+			// User can resume the job if completely ordered with a new sas.
+			if strErr.Response().StatusCode == http.StatusForbidden {
+				errMsg := fmt.Sprintf("Authentication Failed. The SAS is not correct or expired or does not have the correct permission %s", err.Error())
+				jptm.Log(pipeline.LogError, errMsg)
+				common.GetLifecycleMgr().Error(errMsg)
+			}
+		}
+
+		// in all other cases, make the transfer as failed
+		transferDone(common.ETransferStatus.Failed(), err)
+	} else {
+		transferDone(common.ETransferStatus.Success(), nil)
+	}
+}
+
+func setPropertiesFile(jptm IJobPartTransferMgr, p pipeline.Pipeline) {
+	info := jptm.Info()
+	u, _ := url.Parse(info.Source)
+	srcFileURL := azfile.NewFileURL(*u, p)
+
+	// Internal function which checks the transfer status and logs the msg respectively.
+	// Sets the transfer status and Report Transfer as Done.
+	// Internal function is created to avoid redundancy of the above steps from several places in the api.
+	transferDone := func(status common.TransferStatus, err error) {
+		if status == common.ETransferStatus.Failed() {
+			jptm.LogError(info.Source, "SET-PROPERTIES ERROR ", err)
+		} else {
+			jptm.Log(pipeline.LogInfo, fmt.Sprintf("SET-PROPERTIES SUCCESSFUL: %s", strings.Split(info.Destination, "?")[0]))
+		}
+
+		jptm.SetStatus(status)
+		jptm.ReportTransferDone()
+	}
+
+	_, err := srcFileURL.SetMetadata(jptm.Context(), info.SrcMetadata.ToAzFileMetadata())
```

Comment:
Metadata is checked in blob; why not in Files?

Reply:
added